lightning/src/ln/channelmanager.rs

```diff
@@ -3874,12 +3872,9 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 	}
 
 	/// Fails an HTLC backwards to the sender of it to us.
-	/// Note that while we take a channel_state lock as input, we do *not* assume consistency here.
-	/// There are several callsites that do stupid things like loop over a list of payment_hashes
-	/// to fail and take the channel_state lock for each iteration (as we take ownership and may
-	/// drop it). In other words, no assumptions are made that entries in claimable_htlcs point to
-	/// still-available channels.
-	fn fail_htlc_backwards_internal(&self, mut channel_state_lock: MutexGuard<ChannelHolder<Signer>>, source: HTLCSource, payment_hash: &PaymentHash, onion_error: HTLCFailReason, destination: HTLCDestination) {
+	/// Note that while this function pushes events as well as FailHTLC's to fail htlcs for
+	/// designated channels, no assumptions are made that the channels are still available.
+	fn fail_htlc_backwards_internal(&self, source: HTLCSource, payment_hash: &PaymentHash, onion_error: HTLCFailReason,destination: HTLCDestination) {
```

Comment:
Wasn't sure if we'd actually like to send in the `forward_htlc` lock here, to ensure that it isn't drained through `process_pending_htlc_forwards` before all htlcs have been processed in `fail_htlc_backwards`, as well as all htlc mpp fails in `claim_funds`. If we'd actually like that, let me know and I'll take the lock before those loops in `fail_htlc_backwards`/`claim_funds` :).